internal/planner/planner.go

```diff
@@ -299,15 +300,24 @@ func (p *Planner) planRules(rules []*ast.Rule) (string, error) {
 	// Default rules execute if the return is undefined.
 	if defaultRule != nil {
 
-		fn.Blocks = append(fn.Blocks, &ir.Block{
-			Stmts: []ir.Stmt{
-				&ir.IsUndefinedStmt{Source: fn.Return},
-			},
-		})
+		// Set the location for the default rule head.
+		p.loc = defaultRule.Loc()
+		// NOTE(sr) for `default p = 1`,
+		// defaultRule.Loc() is `default`,
+		// defaultRule.Head.Loc() is `p = 1`.
+		// I'd prefer we'd have `default p = 1`; but since the mapped
+		// locations are col/row only (no extent), pointing to `default`
+		// seems preferable.
```

Comment:
SGTM

Reply:
😄 So, overnight, I've kind of convinced myself that this was the wrong choice. I think what matters is the `p = 1`, much more so than the `default`, when it comes to potential errors related to this.

Reply:
I'm fine with either--in practice, I think the row will be the important bit and the answer will be the same.